angularjs-portal-home/src/main/webapp/my-app/layout/widget/controllers.js

```diff
@@ -43,22 +43,38 @@ define(['angular'], function(angular){
     populateWidgetContent();
   }]);
 
-  app.controller('WeatherController', ['$scope', 'layoutService', function($scope, layoutService){
+  app.controller('WeatherController', ['$scope', 'layoutService', 'keyValueService', '$q', function($scope, layoutService, keyValueService, $q){
     $scope.weatherData = [];
     $scope.loading = false;
     $scope.showMetric = false;
     $scope.currentlyImperial = true;
+    $scope.fetchKey="userWeatherPreference";
```

Comment:
Pro tip: Style: use whitespace consistently.  Cf. [Google JavaScript Style Guide](https://google.github.io/styleguide/javascriptguide.xml), noting that the most interesting part of it is not [its specific guidance about this formatting](https://google.github.io/styleguide/javascriptguide.xml?showone=Code_formatting#Code_formatting), but 

> Parting Words

> BE CONSISTENT.

> If you're editing code, take a few minutes to look at the code around you and determine its style. If they use spaces around all their arithmetic operators, you should too. If their comments have little boxes of hash marks around them, make your comments have little boxes of hash marks around them too.

> The point of having style guidelines is to have a common vocabulary of coding so people can concentrate on what you're saying rather than on how you're saying it. We present global style rules here so people know the vocabulary, but local style is also important. If code you add to a file looks drastically different from the existing code around it, it throws readers out of their rhythm when they go to read it. Avoid this.